iib/workers/tasks/opm_operations.py

```diff
@@ -907,3 +907,224 @@ def verify_operator_exists(
             is_package_in_index = True
             log.info("operator package %s found in index_db %s", operator_package, index_db_path)
     return is_package_in_index, index_db_path
+
+
+def set_opm_verison(index_image: str) -> str:
+    """
+    This function will verify if opm config is set in worker
+    if the config entry exists for the index image version
+    and return the opm version path to be used in the entire IIB request.
+    :param str index_image: from_index_image for the request
+    :return str opm_path: opm_path to be used for the entire request
+
+    """
+    global opm_path
+    log.info("Determining wthe OPM version to use")
+    worker_config = get_worker_config()
+    opm_paths_config = worker_config.get("iib_ocp_opm_mapping")
+    if not opm_paths_config:
+            log.error("OPM config is not set")
+    if index_image:
+        index_version =_find_index_version(index_image)
+        for ocp_version, opm_version in opm_paths_config.items():
+            if ocp_version == index_version:
+                opm_path = opm_version
+                break
```

Comment:
I am wondering if simple `opm_path = opm_paths_config.get(index_version, opm_paths_config['default'])` would not be enough? Instead of this for loop.

Reply:
I will try this :+1:

Reply:
Updated